subsys/caf/modules/shell.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/shell/shell.h>
+#include <app_event_manager.h>
+#include <caf/events/button_event.h>
+
+static int set_button_handler(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 3) {
+		shell_error(shell, "Missing required parameter(s)");
+		return -EINVAL;
+	}
+
+	struct button_event *event = new_button_event();
+
+	event->key_id = (uint16_t)atoi(argv[1]);
+	event->pressed = *argv[2] != '0';
+	APP_EVENT_SUBMIT(event);
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_caf_shell_command,
+	SHELL_CMD_ARG(set_button, NULL, "Simulate a button event with button ID and state",
```

Comment:
`Submit a button_event with user-defined key ID and pressed state` ? To make it more in line with naming in CAF.